app/controllers/forms_controller.rb

```diff
@@ -31,4 +36,15 @@ def calculate_next_step(session_data, step)
       Steps::Helper.next_step_for(session_data, step)
     end
   end
+
+  def calculate_early_result
+    # for other groupings, we'll be able to implement like below (disposable, partner gross)
+    # we could also consider a check at each page in the gross income section
+    if last_tag_in_group?(:gross_income)
+      cfe_result = CfeService.result(session_data, Steps::Helper.completed_steps_for(session_data, step))
+      session_data["early_result"] = { "result" => cfe_result.early_gross_income_result,
```

Comment:
we are currently storing in the session:
`early_eligibility_selection` which has a value of `nil`, `gross` or `continued_check`
do you think we need to store this value with this new banner method?

Reply:
So i have actually added a new variable with a slightly different name, that does the same thing - but we don't grab it until later in the flow, in the results controller. I wasn't sure about this naming now we have got rid of the form. I've introduced a new variable called `@early_assessment`.